### DISCUSSION TITLE
How to handle different types with the same short name?

### DISCUSSION THREAD

**Discussion by paul-hansen** [ast-grep/ast-grep#520] [Q&A] [answered]
If two types have the same name, how would I match just one of the types and avoid false positives?

For example, in the code below, if I use the pattern `Clicked` it will match my enum and Bevy's Interaction::Clicked, I want it to just match Bevy's `Interaction::Clicked`.

```rust
use bevy_ui::focus::Interaction;

for mut interaction in button_query.iter_mut(&mut app.world) {
        *interaction = Interaction::Clicked;
    }

pub enum Events{
  Clicked,
  Released,
}
```

Ideally I'd be able to use `bevy_ui::focus::Interaction::Clicked` but that doesn't seem to work.
[Playground](https://ast-grep.github.io/playground.html#eyJtb2RlIjoiUGF0Y2giLCJsYW5nIjoicnVzdCIsInF1ZXJ5IjoiQ2xpY2tlZCIsInJld3JpdGUiOiJQcmVzc2VkIiwiY29uZmlnIjoiIyBZQU1MIFJ1bGUgaXMgbW9yZSBwb3dlcmZ1bCFcbiMgaHR0cHM6Ly9hc3QtZ3JlcC5naXRodWIuaW8vZ3VpZGUvcnVsZS1jb25maWcuaHRtbCNydWxlXG5ydWxlOlxuICBhbnk6XG4gICAgLSBwYXR0ZXJuOiBcIiRBUFAuYWRkX3N5c3RlbSgkJCRGT08pXCJcbiAgICAtIHBhdHRlcm46IFwiJEFQUC5hZGRfc3lzdGVtKCQkJEZPTy5pbl9zY2hlZHVsZShVcGRhdGUpKVwiXG5maXg6XG4gIFwiJEFQUC5hZGRfc3lzdGVtcyhVcGRhdGUsICQkJEZPTylcIiIsInNvdXJjZSI6InVzZSBiZXZ5X3VpOjpmb2N1czo6SW50ZXJhY3Rpb247XG5cbmZvciBtdXQgaW50ZXJhY3Rpb24gaW4gYnV0dG9uX3F1ZXJ5Lml0ZXJfbXV0KCZtdXQgYXBwLndvcmxkKSB7XG4gICAgICAgICppbnRlcmFjdGlvbiA9IEludGVyYWN0aW9uOjpDbGlja2VkO1xuICAgIH1cblxucHViIGVudW0gRXZlbnRze1xuICBDbGlja2VkLFxuICBSZWxlYXNlZCxcbn0ifQ==)

**Comment by HerringtonDarkholme** [comment, marked as answer]
I'm sorry but ast-grep at the moment is a syntax only tool. It cannot resolve the identifier type now.

You can however using `--interactive` flag in the CLI tool to select the right change manually.
